invenio.cfg

```diff
@@ -176,4 +176,104 @@ OAUTHCLIENT_AUTO_REDIRECT_TO_EXTERNAL_LOGIN = False  # autoredirect to external
 # --------------------
 USERPROFILES_READ_ONLY = False  # allow users to change profile info (name, email, etc...)
 
-THEME_SHOW_FRONTPAGE_INTRO_SECTION = False
\ No newline at end of file
+THEME_SHOW_FRONTPAGE_INTRO_SECTION = False
+
+# Custom fields
+# -------------
+from invenio_records_resources.services.custom_fields import KeywordCF, BaseCF
+from marshmallow import fields
+from marshmallow_utils.fields import SanitizedUnicode
+
+
+class RelationshipListCF(BaseListCF):
+    """Relationship list custom field."""
+
+    @property
+    def mapping(self):
+        return {
+            # NOTE: "object" wouldn't work for what we're querying
+            "type": "nested",
+            "properties": {
+                "subject": {"type": "keyword"},
+                "object": {"type": "keyword"},
+            },
+        }
+
+    @property
+    def field(self):
+        # For example see "Host of" on https://zenodo.org/record/3949282
+        return fields.List(fields.Nested({
+            'subject': fields.List(SanitizedUnicode(required=True)),
+            'object': fields.List(SanitizedUnicode(required=True)),
+        }))
+
+
+
+RDM_NAMESPACES = {
+    "dwc": "http://rs.tdwg.org/dwc/terms/",
+    "gbif-dwc": "http://rs.gbif.org/terms/1.0/",
+    "ac": "http://rs.tdwg.org/ac/terms/",
+    "openbiodiv": "http://openbiodiv.net/ontology#",
+    "obo": "http://purl.obolibrary.org/obo/",
+    "dc": "http://purl.org/dc/terms/",
+}
+
+
+RDM_CUSTOM_FIELDS = [
+    # dwc
+    KeywordCF(name='dwc:basisOfRecord', multiple=True),
+    KeywordCF(name='dwc:catalogNumber', multiple=True),
+    KeywordCF(name='dwc:class', multiple=True),
+    KeywordCF(name='dwc:collectionCode', multiple=True),
+    KeywordCF(name='dwc:country', multiple=True),
+    KeywordCF(name='dwc:county', multiple=True),
+    KeywordCF(name='dwc:dateIdentified', multiple=True),
```

Comment:
minor: would it be useful to make them iso or edtf fields? that would allow for example to do range queries on them

Reply:
Agree. I'm only afraid because they're not validated at the moment, so we might shoot ourselves in the foot when doing the migration (but it's worth a try).